app/src/main/java/com/example/Egida/presentation/ui/RegistrationFragment.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.Egida.presentation.ui
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import androidx.core.widget.doAfterTextChanged
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import com.example.Egida.R
+import com.example.Egida.presentation.viewModel.MainViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlin.properties.Delegates
+
+class RegistrationFragment : Fragment() {
+
+    companion object {
+        fun newInstance() = RegistrationFragment()
+        const val TAG = " registrationFragment"
+    }
+
+    private var fragment = R.layout.registration_fragment
+    private var chek by Delegates.notNull<Boolean>()
```

Comment:
Do not get this delegate logic. Will be nice, if you write me a message with a description of reasons to do this.